Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/settings/SettingsFragmentPresenter.java

```diff
@@ -179,8 +180,28 @@ private void addCoreSettings(ArrayList<SettingsItem> sl)
 			mView.passSettingsToActivity(mSettings);
 		}
 
-		// TODO Set default value for cpuCore based on arch.
-		sl.add(new SingleChoiceSetting(SettingsFile.KEY_CPU_CORE, SettingsFile.SECTION_CORE, SettingsFile.SETTINGS_DOLPHIN, R.string.cpu_core, 0, R.array.emuCoresEntries, R.array.emuCoresValues, 4, cpuCore));
+		// TODO: Having different emuCoresEntries/emuCoresValues for each architecture is annoying.
```

Comment:
IIRC, the reason we are relying on different `R.array` resources depending on arch is that we were originally using an off-the-shelf settings UI android provides, and when using that, `R.array`s are how you provide options. We aren't doing that anymore, so with a little refactoring we could construct the list without them.